components/task/details/solutions/newSolutionForm/NewSolutionForm.tsx

```diff
@@ -0,0 +1,139 @@
+import React from 'react';
+import Button, { ButtonLayout } from '../../../../shared/button/Button';
+import Select from 'react-select';
+import { formStyles } from './formStyles';
+import clsx from 'clsx';
+import NewSolutionFormInput from './newSolutionFormInput/NewSolutionFormInput';
+import { Form, Field } from 'react-final-form';
+
+export const technologies = [
+  { value: 'html', label: '#html' },
+  { value: 'css', label: '#css' },
+  { value: 'js', label: '#js' },
+  { value: 'nodejs', label: '#nodejs' },
+  { value: 'python', label: '#python' },
+  { value: 'react', label: '#react' },
+  { value: 'vue', label: '#vue' },
+  { value: 'angular', label: '#angular' },
+];
+
+interface Technology {
+  value: string;
+  label: string;
+}
+
+interface NewSolutionFormProps {
+  setIsModalOpen: React.Dispatch<React.SetStateAction<boolean>>;
+}
+
+const NewSolutionForm: React.FC<NewSolutionFormProps> = ({ setIsModalOpen }) => {
+  const handleSubmit = (values: string[]): void => {
+    console.log(JSON.stringify(values));
```

Comment:
to musi zwracać promise / lub error:
nazwij tę funkcję inaczej niż handleSubmit, np. onSubmit:
czyli:
```ts
interface FormValues {
  solutionLinkInput: string;
  solitionXXX: string;
  smth: boolean;
}
const onSubmit = (values: FormValues) => {
  return sendSolutions(values)
    .then( data => onSucess)
    .catch( error => {
      return error.message
    })
```
koniecznie w catch musisz zwrócić return string -> który w Form będzie pod propsem: meta.submitError

Reply:
Zmieniłem nazwę i przerobiłem, ale nie mam pojęcia czy zrobiłem to dobrze... :(